### ISSUE TITLE
timestamp precision issue

### ISSUE THREAD

**Issue by eddelbuettel** [eddelbuettel/rcppmsgpack#11]
@traversc: I had this for me once at home, and once at Travis.  I think `identical` is too strict:

```r
> library(RcppMsgPack)
> mt <- Sys.time()
> attr(mt, "tzone") <- "UTC"
> mp <- msgpack_pack(msgpack_timestamp_encode(mt))
> mt <- Sys.time()
> attr(mt, "tzone") <- "UTC"
> mp <- msgpack_pack(msgpack_timestamp_encode(mt))
> mtu <- msgpack_timestamp_decode(msgpack_unpack(mp))
> stopifnot(identical(mt, mtu))
Error: identical(mt, mtu) is not TRUE
> mt
[1] "2018-01-13 03:04:21 UTC"
> mtu
[1] "2018-01-13 03:04:21 UTC"
> mt - mtu
Time difference of -2.384186e-07 secs
> all.equal(mt,mtu)
[1] TRUE
> 
```

`all.equal` would do. For `POSIXct` the resolution is actually less that -2.3e-07 as I recall.

(Another set of changes is just make the compiler a little more silent.)

**Comment by eddelbuettel** [comment]
I will take care of both changes and will ping you when the PR is ready, ok?

**Comment by traversc** [comment]
Thank makes sense, thanks for catching that.

**Comment by eddelbuettel** [comment]
Fixed in #12
